aws/dev.yml

```diff
@@ -47,29 +47,20 @@
 
     - debug: var=stack
 
-    - debug: msg={{ stack | json_query('stack_resources[?logical_resource_id==`FxaELB`].physical_resource_id | [0]') }}
-
-    - name: Waiting for instance to join ELB (ignore FAILED RETRYING log messages)
-      ec2_elb_facts:
-        names:
-          - "{{ stack | json_query('stack_resources[?logical_resource_id==`FxaELB`].physical_resource_id | [0]') }}"
-        region: "{{ region }}"
-      ignore_errors: yes
-      register: fxaelb
-      until: >
-        fxaelb and fxaelb.get('elbs') and (fxaelb.get('elbs') | length) > 0
-          and fxaelb.get('elbs')[0].get('instances_inservice_count')
-          and fxaelb.get('elbs')[0].get('instances_inservice_count') == 1
+    - name: Wait for nginx on the instance to come online (ignore FAILED RETRYING log messages)
+      uri:
+        url: "http://meta-{{ stack['stack_outputs']['ContentHost'] }}:1443/50x.html"
+        follow_redirects: none
+        status_code:
+          - -1
+          - 200
+          - 404
+      register: response
+      until: response and response.get('status') != -1
```

Comment:
I changed this to just poll the instance waiting for nginx to come up, which actually happens slightly before the instance is able to join the LB, because there's a period of time where nginx is up and running but isn't serving the right pages for the LB healthchecks to pass. Combined with the loop below that checks for __version__ files on the various endpoints, I think it gives a good sense of progress of the setup.